_getline.c

```diff
@@ -0,0 +1,52 @@
+#include "shell.h"
+
+/**
+ * _getline - reads an entire line from fd, storing the address of
+ *the buffer containing the text into *lineptr.
+ * @fd: the stream.
+ * @lineptr: the buffer containing the line from fd.
+ * @n: the size of the buffer.
+ *
+ * Return: number of characters read, or -1 on failure.
+ */
+
+ssize_t _getline(char **lineptr, size_t *n, int fd)
+{
+	char *new_ptr;
+	static size_t count;
```

Comment:
i didn't understand you point about the `static var`

Reply:
I think you may understand from this pseudo code:
```c
int _getline(char *line, const fd)
{
    int r, pos;
    static char *rd = NULL; // content read
    char *tmp;
    char buff[BUFF_SIZE + 1];
  
    // Step 1 can be wrapped into function
    pos = str_find(rd, '\n'); // if rd is NULL or character not found str_find return -1 otherwise position of char
    if (pos >= 0)
    {
         *line = str_copy(rd, pos ); // copy content until newline
         tmp = rd;
         rd = strdup(rd + pos + 1);
         free(tmp);
         return(pos);
    }
    while ((r = read(fd, buff ,BUFF_SIZE)) != 0)
    {
         if (r < 0) return (-1);
         buff[r] = '\0';
         strjoin(rd, buff);
         if (str_find('\n', buff) || r < BUFF_SIZE)
              break;
    }
     pos =  str_find(rd, '\n'); // or str_find(rd, EOF);
     *line = str_copy(rd, pos ); // copy content until newline
     tmp = rd;
     rd = strdup(rd + pos + 1);
     free(tmp);
     return(pos);
}
```